### ISSUE TITLE
Compilation warnings

### ISSUE THREAD

**Issue by Lecrapouille** [SFML/SFML#2744]
### Prerequisite Checklist

- [X] I searched for [existing issues](https://github.com/search?q=repo%3ASFML%2FSFML&type=issues) to prevent duplicates
- [X] I searched for [existing discussions on the forum](https://www.google.com/search?q=site%3Ahttps%3A%2F%2Fen.sfml-dev.org) to prevent duplicates
- [X] I am here to report an issue and not to just ask a question or look for help (use the [forum](https://en.sfml-dev.org/forums/index.php#c3) or [Discord](https://discord.gg/nr4X7Fh) instead)

### Describe your issue here

I can produce warnings that with -Werror will halt the compilation from GLCheck.hpp and OpenGL_ES when compiling with -DSFML_OPENGL_ES=TRUE -DSFML_USE_DRM=FALSE

### Your Environment

- OS / distro / window manager: Debian 11
- SFML version: 2.6.0 (SHA1: 11b73743c42cf7ecd7c596ba83fdbf1150ffa96c)
- Compiler / toolchain: g++-10
- Special compiler / CMake flags: yes cmake -DWARNINGS_AS_ERRORS=OFF to allow compiling

### Steps to reproduce

```bash
mkdir build
cd build

cmake -DSFML_BUILD_AUDIO=FALSE -DSFML_BUILD_DOC=FALSE -DSFML_BUILD_EXAMPLES=FALSE -DSFML_BUILD_NETWORK=FALSE -DSFML_BUILD_TEST_SUITE=FALSE -DSFML_BUILD_WINDOW=TRUE -DSFML_OPENGL_ES=TRUE -DSFML_USE_DRM=FALSE -DBUILD_SHARED_LIBS=OFF -DWARNINGS_AS_ERRORS=OFF ..

make -j8
```

will produce errors:

```
In file included from src/SFML/Graphics/Texture.cpp:30:
src/SFML/Graphics/Texture.cpp: In member function ‘bool sf::Texture::create(unsigned int, unsigned int)’:
src/SFML/Graphics/Texture.cpp:210:108: warning: this condition has identical branches [-Wduplicated-branches]
  210 |     glCheck(glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_S, m_isRepeated ? GL_REPEAT : (textureEdgeClamp ? GLEXT_GL_CLAMP_TO_EDGE : GLEXT_GL_CLAMP)));
src/SFML/Graphics/GLCheck.hpp:51:28: note: in definition of macro ‘glCheck’
   51 |     #define glCheck(expr) (expr)
      |                            ^~~~
src/SFML/Graphics/Texture.cpp:211:108: warning: this condition has identical branches [-Wduplicated-branches]
  211 |     glCheck(glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_T, m_isRepeated ? GL_REPEAT : (textureEdgeClamp ? GLEXT_GL_CLAMP_TO_EDGE : GLEXT_GL_CLAMP)));
src/SFML/Graphics/GLCheck.hpp:51:28: note: in definition of macro ‘glCheck’
   51 |     #define glCheck(expr) (expr)
      |                            ^~~~
src/SFML/Graphics/Texture.cpp: In member function ‘sf::Image sf::Texture::copyToImage() const’:
src/SFML/Graphics/Texture.cpp:349:43: warning: conversion to ‘GLsizei’ {aka ‘int’} from ‘unsigned int’ may change the sign of the result [-Wsign-conversion]
  349 |         glCheck(glReadPixels(0, 0, m_size.x, m_size.y, GL_RGBA, GL_UNSIGNED_BYTE, &pixels[0]));
      |                                    ~~~~~~~^
src/SFML/Graphics/GLCheck.hpp:51:28: note: in definition of macro ‘glCheck’
   51 |     #define glCheck(expr) (expr)
      |                            ^~~~
src/SFML/Graphics/Texture.cpp:349:53: warning: conversion to ‘GLsizei’ {aka ‘int’} from ‘unsigned int’ may change the sign of the result [-Wsign-conversion]
  349 |         glCheck(glReadPixels(0, 0, m_size.x, m_size.y, GL_RGBA, GL_UNSIGNED_BYTE, &pixels[0]));
      |                                              ~~~~~~~^
src/SFML/Graphics/GLCheck.hpp:51:28: note: in definition of macro ‘glCheck’
   51 |     #define glCheck(expr) (expr)
      |                            ^~~~
src/SFML/Graphics/Texture.cpp:352:63: warning: conversion to ‘GLuint’ {aka ‘unsigned int’} from ‘GLint’ {aka ‘int’} may change the sign of the result [-Wsign-conversion]
  352 |         glCheck(GLEXT_glBindFramebuffer(GLEXT_GL_FRAMEBUFFER, previousFrameBuffer));
      |                                                               ^~~~~~~~~~~~~~~~~~~
src/SFML/Graphics/GLCheck.hpp:51:28: note: in definition of macro ‘glCheck’
   51 |     #define glCheck(expr) (expr)
      |                            ^~~~
src/SFML/Graphics/Texture.cpp: In member function ‘void sf::Texture::setRepeated(bool)’:
src/SFML/Graphics/Texture.cpp:680:116: warning: this condition has identical branches [-Wduplicated-branches]
  680 |             glCheck(glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_S, m_isRepeated ? GL_REPEAT : (textureEdgeClamp ? GLEXT_GL_CLAMP_TO_EDGE : GLEXT_GL_CLAMP)));
src/SFML/Graphics/GLCheck.hpp:51:28: note: in definition of macro ‘glCheck’
   51 |     #define glCheck(expr) (expr)
      |                            ^~~~
src/SFML/Graphics/Texture.cpp:681:116: warning: this condition has identical branches [-Wduplicated-branches]
  681 |             glCheck(glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_T, m_isRepeated ? GL_REPEAT : (textureEdgeClamp ? GLEXT_GL_CLAMP_TO_EDGE : GLEXT_GL_CLAMP)));
src/SFML/Graphics/GLCheck.hpp:51:28: note: in definition of macro ‘glCheck’
   51 |     #define glCheck(expr) (expr)
      |                            ^~~~

In file included from src/SFML/Graphics/Texture.cpp:30:
src/SFML/Graphics/Texture.cpp: In member function ‘bool sf::Texture::create(unsigned int, unsigned int)’:
src/SFML/Graphics/Texture.cpp:210:108: warning: this condition has identical branches [-Wduplicated-branches]
  210 |     glCheck(glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_S, m_isRepeated ? GL_REPEAT : (textureEdgeClamp ? GLEXT_GL_CLAMP_TO_EDGE : GLEXT_GL_CLAMP)));
src/SFML/Graphics/GLCheck.hpp:51:28: note: in definition of macro ‘glCheck’
   51 |     #define glCheck(expr) (expr)
      |                            ^~~~
src/SFML/Graphics/Texture.cpp:211:108: warning: this condition has identical branches [-Wduplicated-branches]
  211 |     glCheck(glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_T, m_isRepeated ? GL_REPEAT : (textureEdgeClamp ? GLEXT_GL_CLAMP_TO_EDGE : GLEXT_GL_CLAMP)));
src/SFML/Graphics/GLCheck.hpp:51:28: note: in definition of macro ‘glCheck’
   51 |     #define glCheck(expr) (expr)
      |                            ^~~~
src/SFML/Graphics/Texture.cpp: In member function ‘sf::Image sf::Texture::copyToImage() const’:
src/SFML/Graphics/Texture.cpp:349:43: warning: conversion to ‘GLsizei’ {aka ‘int’} from ‘unsigned int’ may change the sign of the result [-Wsign-conversion]
  349 |         glCheck(glReadPixels(0, 0, m_size.x, m_size.y, GL_RGBA, GL_UNSIGNED_BYTE, &pixels[0]));
      |                                    ~~~~~~~^
src/SFML/Graphics/GLCheck.hpp:51:28: note: in definition of macro ‘glCheck’
   51 |     #define glCheck(expr) (expr)
      |                            ^~~~
src/SFML/Graphics/Texture.cpp:349:53: warning: conversion to ‘GLsizei’ {aka ‘int’} from ‘unsigned int’ may change the sign of the result [-Wsign-conversion]
  349 |         glCheck(glReadPixels(0, 0, m_size.x, m_size.y, GL_RGBA, GL_UNSIGNED_BYTE, &pixels[0]));
      |                                              ~~~~~~~^
src/SFML/Graphics/GLCheck.hpp:51:28: note: in definition of macro ‘glCheck’
   51 |     #define glCheck(expr) (expr)
      |                            ^~~~
src/SFML/Graphics/Texture.cpp:352:63: warning: conversion to ‘GLuint’ {aka ‘unsigned int’} from ‘GLint’ {aka ‘int’} may change the sign of the result [-Wsign-conversion]
  352 |         glCheck(GLEXT_glBindFramebuffer(GLEXT_GL_FRAMEBUFFER, previousFrameBuffer));
      |                                                               ^~~~~~~~~~~~~~~~~~~
src/SFML/Graphics/GLCheck.hpp:51:28: note: in definition of macro ‘glCheck’
   51 |     #define glCheck(expr) (expr)
      |                            ^~~~
src/SFML/Graphics/Texture.cpp: In member function ‘void sf::Texture::setRepeated(bool)’:
src/SFML/Graphics/Texture.cpp:680:116: warning: this condition has identical branches [-Wduplicated-branches]
  680 |             glCheck(glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_S, m_isRepeated ? GL_REPEAT : (textureEdgeClamp ? GLEXT_GL_CLAMP_TO_EDGE : GLEXT_GL_CLAMP)));
src/SFML/Graphics/GLCheck.hpp:51:28: note: in definition of macro ‘glCheck’
   51 |     #define glCheck(expr) (expr)
      |                            ^~~~
src/SFML/Graphics/Texture.cpp:681:116: warning: this condition has identical branches [-Wduplicated-branches]
  681 |             glCheck(glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_T, m_isRepeated ? GL_REPEAT : (textureEdgeClamp ? GLEXT_GL_CLAMP_TO_EDGE : GLEXT_GL_CLAMP)));
src/SFML/Graphics/GLCheck.hpp:51:28: note: in definition of macro ‘glCheck’
   51 |     #define glCheck(expr) (expr)
      |                            ^~~~
```

### Expected behavior

No warnings since compiled with -Werror

### Actual behavior

Undesired warnings

**Comment by ChrisThrasher** [comment]
https://github.com/SFML/SFML/pull/2680

The good news is that warnings-as-errors is disabled by default in the soon-to-be-released SFML 2.6.1 for precisely this reason. We don't want users having to disable that option just to get the library to build. 

Is this problem limited to only when `SFML_OPENGL_ES` is enabled? Can you build with that option disabled to see if the warning persists?

**Comment by ChrisThrasher** [comment]
https://github.com/SFML/SFML/actions/runs/6620148970/job/17982019028

I can reproduce this when enabling an OpenGL ES job.

**Comment by ChrisThrasher** [comment]
@Lecrapouille please try out https://github.com/SFML/SFML/pull/2747 to confirm it fixes you issue

**Comment by Lecrapouille** [comment]
@ChrisThrasher yes only when SFML_OPENGL_ES is enabled. And you commit fixed warnings.

**Comment by ChrisThrasher** [comment]
Fixed by #2747`
